### PR TITLE
Polishing che-core-commons-schedule

### PR DESCRIPTION
**Pull request by alexVengrovsk**
Signed-off-by: alextrentton@gmail.com

### CODE REVIEW

**Comment by alexVengrovsk** [comment]
@skabashnyuk I also have some doubts about fielsd and methods scopes in this class. Many of them have _protected_  access modifiers, but could be changed to private. Let me know is it OK.

**Comment by alexVengrovsk** [comment]
@skabashnyuk One more thing, the _findNextWhiteSpace()_ method doesn't work as it should. Is it need to be fixed?

**Comment by garagatyi** [comment]
Please do not use wildcard imports. We don't use it.

File: core/commons/che-core-commons-schedule/src/main/java/org/eclipse/che/commons/schedule/executor/CronExpression.java [hunk 1]
```diff
@@ -846,101 +840,35 @@ public String getCronExpression() {
...
-        buf.append("\n");
-        buf.append("lastdayOfMonth: ");
-        buf.append(lastdayOfMonth);
-        buf.append("\n");
-        buf.append("years: ");
-        buf.append(getExpressionSetSummary(years));
-        buf.append("\n");
-
-        return buf.toString();
+        return "seconds: " + getExpressionCollectionSummary(seconds) + "\n"
```
Comment (garagatyi) [inline core/commons/che-core-commons-schedule/src/main/java/org/eclipse/che/commons/schedule/executor/CronExpression.java:852]: Can you explain why you've changed that? AFAIK usage of StringBuilder is more convinient

**Comment by alexVengrovsk** [comment]
@garagatyi Ok. I will change this wildcard imports. Is it something else wrong in my PR?

File: core/commons/che-core-commons-schedule/src/main/java/org/eclipse/che/commons/schedule/executor/CronExpression.java [hunk 2]
```diff
@@ -846,101 +840,35 @@ public String getCronExpression() {
...
-                buf.append(",");
-            }
-            buf.append(val);
-            first = false;
-        }
-
-        return buf.toString();
+        return collection.stream()
+                .map(Object::toString)
+                .collect(Collectors.joining(","));
```
Comment (garagatyi) [inline core/commons/che-core-commons-schedule/src/main/java/org/eclipse/che/commons/schedule/executor/CronExpression.java:876]: Yeah, stream looks much better here!

File: core/commons/che-core-commons-schedule/src/main/java/org/eclipse/che/commons/schedule/executor/CronExpression.java (hunk 1 above)
Comment (alexVengrovsk) [inline core/commons/che-core-commons-schedule/src/main/java/org/eclipse/che/commons/schedule/executor/CronExpression.java:852]: @garagatyi I did it becouse string concatenation will be represented by compiler with a help os StringBuilder any way. So, it seams to me, that it was unnecessary code (creation StringBuilder object and all _append()_ methods and _toString()_ method).

**Comment by alexVengrovsk** [comment]
@garagatyi Here is one more thing about this class, but I am not shure about this.

```
public static boolean isValidExpression(String cronExpression) {

        try {
            new CronExpression(cronExpression);
        } catch (ParseException pe) {
            return false;
        }

        return true;
    }
```

This kind of check is very coastly. Like we need to ceate whole object here. Maybe it is a good idea to think how to improve it?

**Comment by alexVengrovsk** [comment]
@garagatyi  I have chended import block.

**Comment by skabashnyuk** [comment]
Hello @myArea51 
Is this pr add some new feature. Or this is more code cleanup?
You might not notice but this code originally from quartz. I would like not to introduce changes without strong need.

**Comment by alexVengrovsk** [comment]
Hi @skabashnyuk!  This PR is more about polishing. But take a close look at  the _findNextWhiteSpace()_ method. I have changed it, becouse in its previous state it worked wrong. Instead of returning an index of the next whitespace it returned a string length from the start position to the end of a string. I dont think, that it is ok. But it is up to you to decide.

**Comment by skabashnyuk** [comment]
Can you do pr to this file https://github.com/quartz-scheduler/quartz/blob/master/quartz-core/src/main/java/org/quartz/CronExpression.java first

**Comment by alexVengrovsk** [comment]
@skabashnyuk Ok. I will do it. So, do I needto close this one?

**Comment by skabashnyuk** [comment]
Yes I think. When quartz approve your changes. We can accept this changes back with new pr.

**Comment by alexVengrovsk** [comment]
Ok @skabashnyuk

**Pull request closed without merging**